payments-core/src/main/java/com/stripe/android/cards/RemoteCardAccountRangeSource.kt

```diff
@@ -26,23 +25,20 @@ internal class RemoteCardAccountRangeSource(
         cardNumber: CardNumber.Unvalidated
     ): List<AccountRange>? {
         return cardNumber.bin?.let { bin ->
-            _loading.value = true
+            val accountRanges = withLoading {
+                stripeRepository.getCardMetadata(bin, requestOptions)?.accountRanges
+            }
 
-            val accountRanges =
-                stripeRepository.getCardMetadata(bin, requestOptions)?.accountRanges.orEmpty()
-            cardAccountRangeStore.save(bin, accountRanges)
 
-            _loading.value = false
 
-            when {
-                accountRanges.isNotEmpty() -> {
-                    if (cardNumber.isValidLuhn) {
-                        onCardMetadataMissingRange()
-                    }
-                    accountRanges
+            if (!accountRanges.isNullOrEmpty()) {
+                val hasMatch = accountRanges.any { it.binRange.matches(cardNumber) }
+                if (!hasMatch && cardNumber.isValidLuhn) {
```

Comment:
Re-adding the `hasMatch` check that was removed in https://github.com/stripe/stripe-android/pull/6335/files#diff-d6f7381dc56080adb96df46f224f39adff555b81bdff2952944a80530f1d9e03L39.

cc @jameswoo-stripe @davidme-stripe @wooj-stripe: I’m not fully sure if this implementation is correct. It seems _more_ correct than before, but iOS does have some additional logic that I don’t understand. Any thoughts?

Reply:
Sorry, I don't remember why I removed that piece of logic :( It does seem odd that it was removed!